src/mbed_tools/cli/build.py

```diff
@@ -124,13 +124,6 @@ def build(
         terminal.run(dev.serial_port, baudrate)
 
 
-def _validate_target_and_toolchain_args(target: str, toolchain: str) -> None:
-    if not all([toolchain, target]):
-        raise click.UsageError(
-            "Both --toolchain and --mbed-target arguments are required when using the compile subcommand."
-        )
-
-
```

Comment:
The idea of removing this is to
* Take advantage of `required=True` of "click"
* Avoid having to set default arguments (to `""`) in `build()`

The slight drawback is "click" only checks one argument at a time. But eventually a user will need to pass both `--toolchain` and `--mbed-target` to not get any errors, so the outcome is the same.

This is also consistent with the `configure()` command.

Reply:
Please ensure this comment's info is in the commit message that makes this change. It's okay for commit messages to be a bit long.

Reply:
From the current commit message:
```
We set default values of arguments with "click", so there is no point
for `build()` to have default arguments again. Make `--toolchain`
and `--mbed-targets` required, to guarantee the variables exist.
```
Is this enough? Would you suggest also suggest mentiong the "drawback" and "consistency with configure()"?

Reply:
Yes, drawbacks and consistency should be mentioned in the commit message, too. It helps gives better context around what the developer (you) was thinking at the time and what tradeoffs were made.

You felt you needed to comment in the PR beyond what was in the commit message. I am happy if you are satisfied the commit message containing all the information you wanted to share in the comment.

Reply:
> You felt you needed to comment in the PR beyond what was in the commit message.

Good point, I think the tradeoff is worth mentioning in this case. Will update the commit message.